microsoft/testsuites/vm_extensions/azure_disk_encryption.py

```diff
@@ -0,0 +1,221 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import json
+import string
+import time
+from typing import Any
+
+from assertpy import assert_that
+from azure.mgmt.keyvault.models import AccessPolicyEntry, Permissions
+from azure.mgmt.keyvault.models import Sku as KeyVaultSku
+from azure.mgmt.keyvault.models import VaultProperties
+
+from lisa import Logger, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import CBLMariner, CentOs, Oracle, Redhat, Ubuntu
+from lisa.sut_orchestrator import AZURE
+from lisa.sut_orchestrator.azure.common import (
+    AzureNodeSchema,
+    create_keyvault,
+    get_identity_id,
+    get_matching_key_vault_name,
+    get_tenant_id,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform, AzurePlatformSchema
+from lisa.testsuite import TestResult, simple_requirement
+from lisa.util import SkippedException, generate_random_chars
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="Tests for the Azure Disk Encryption (ADE) extension",
+    requirement=simple_requirement(
+        min_memory_mb=8 * 1024,
+        supported_features=[AzureExtension],
+        supported_platform_type=[AZURE],
+    ),
+)
+class AzureDiskEncryption(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node = kwargs["node"]
+        if not self._is_supported_linux_distro(node):
+            raise SkippedException(
```

Comment:
Raise it like below, so the os information can be bring out automatically.

`raise SkippedException(UnsupportedDistroException(node.os))`

Reply:
Updated, included support for min supported versions as well.